src/modbus_client.py

```
@@ -0,0 +1,526 @@
+        if self.client:
+            try:
+                self.client.close()
+                _logger.info("Disconnected from Waveshare gateway")
+            except Exception as e:
+                _logger.error(f"Error disconnecting: {e}")
+            finally:
+                self.connected = False
+                self.client = None
+    
+    async def write_zone_setpoint(self, base_id: int, zone_id: int, 
+                                  dpt_9001_value: int, max_retries: int = None) -> Tuple[bool, str]:
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)